File: C#/0118. Pascal's Triangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {       
        IList<IList<int>> nums=new List<IList<int>>();
        if (numRows==0){
            return nums;
        }
        nums.Add(new List<int>(){1});
        if(numRows==1){
            return nums;
        }
        IList<int> prenums= new List<int>(){1,1};
        nums.Add(prenums);
            
        for(int i=3;i<numRows+1;i++){
            IList<int> curnums=new List<int>{1};
            for(int j=0;j<prenums.Count-1;j++){
                curnums.Add(prenums[j]+prenums[j+1]);
            }
            curnums.Add(1);
            nums.Add(curnums);
            prenums=curnums;
        }
        
        return nums;
        
        
    }
}

File: archives/20190519python/0118. Pascal's Triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        rep=[]
        if numRows==0:
            return rep
        rep.append([1])
        if numRows==1:
            return rep
        rep.append([1,1])
        if numRows==2:
            return rep
        while(numRows>2):
            prev=rep[-1]
            cur=[1]
            for k in range(len(prev)-1):
                cur.append(prev[k]+prev[k+1])
            cur.append(1)
            rep.append(cur)
            numRows-=1
        return rep
            
